### DISCUSSION TITLE
Runtime warnings for non-main thread access broken?

### DISCUSSION THREAD

**Discussion by lukeredpath** [pointfreeco/swift-composable-architecture#1243] [General]
I've just discovered a bug in our code where our wrapper around the Apple `LinkPresentation` framework was completing an effect (using `.future`) on a background thread. This happens quite frequently as we load link previews but even though I can hit a break point on the `runtimeWarning` call in the store's `threadCheck` method, I am not seeing any runtime warnings emitted in Xcode. I am using Xcode 13.3. Has something broken?

**Comment by mbrandonw** [comment]
We've seen warnings sometimes be delayed a bit, but they still seem to consistently show for us. Could you try in a different project (like maybe even a TCA case study) and see if it happens there too?
